hphp/runtime/vm/jit/vasm-ppc64.cpp

```diff
@@ -239,6 +239,24 @@ struct Vgen {
   void emit(const landingpad&) { }
   void emit(const ldimmw& i) { a.li(Reg64(i.d), i.s); }
   void emit(const ldarx& i) { a.ldarx(i.d, i.s); }
+  void emit(const loadups& i) {
+    // A lowering would be handy here, but the Vptr's displacement is only
+    // later added at the optimizeCopies optimization
+    Vptr p(i.s);
+    if ((!p.base.isValid()) || (p.base == 0)) {
+      p.base = ppc64_asm::reg::r0; // use zero not r0
+    }
+    if (!p.index.isValid()) {
+      a.li(rAsm, p.disp);
+      p.index = rAsm;
+      p.disp = 0;
+    } else if (p.disp) {
+        a.addi(rAsm, p.index, p.disp);
+        p.index = rAsm;
+        p.disp = 0;
+    }
+    a.lxvd2x(i.d, p);
+  }
```

Comment:
The `fixVptr` changes the `Vptr` to have base and index or base and displacement. The `loadups` and `storeups` work only with base and index, so in this case we had to add this guard, as only these two instructions are different.